modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java

```diff
@@ -485,530 +424,6 @@ public void setComponentType(TypedComponentIdentifier.ComponentType componentTyp
-    extensionModelHelper.walkToComponent(getIdentifier(), new ExtensionWalkerModelDelegate() {
-
-      @Override
-      public void onConfiguration(ConfigurationModel model) {
-        setConfigurationModel(model);
-        onParameterizedModel(model);
-      }
-
-      @Override
-      public void onConnectionProvider(ConnectionProviderModel model) {
-        setConnectionProviderModel(model);
-        onParameterizedModel(model);
-      }
-
-      @Override
-      public void onOperation(OperationModel model) {
-        setComponentModel(model);
-        onParameterizedModel(model);
-      }
-
-      @Override
-      public void onSource(SourceModel model) {
-        setComponentModel(model);
-        onParameterizedModel(model);
-      }
-
-      @Override
-      public void onConstruct(ConstructModel model) {
-        setComponentModel(model);
-        onParameterizedModel(model);
-      }
-
-      @Override
-      public void onNestableElement(NestableElementModel model) {
-        setNestableElementModel(model);
-        if (model instanceof ParameterizedModel) {
-          onParameterizedModel((ParameterizedModel) model);
-        }
-      }
-
-      private void onParameterizedModel(ParameterizedModel model) {
-        if (runtimeMode) {
-          handleNestedParametersWithoutPopulatingObjectTypes(ComponentModel.this, extensionModelHelper, model);
-        } else {
-          DslElementSyntax elementDsl = extensionModelHelper.resolveDslElementModel(model, getIdentifier());
-          onParameterizedModel(elementDsl, model, parameterModel -> true);
-        }
-      }
-
-      private void onParameterizedModel(DslElementSyntax elementDsl, ParameterizedModel model,
-                                        Predicate<ParameterModel> parameterModelFilter) {
-        Multimap<ComponentIdentifier, ComponentModel> nestedComponents = getNestedComponents(ComponentModel.this);
-
-        List<ParameterModel> inlineGroupedParameters = model.getParameterGroupModels().stream()
-            .filter(ParameterGroupModel::isShowInDsl)
-            .map(group -> addInlineGroup(elementDsl, nestedComponents, group, extensionModelHelper))
-            .flatMap(g -> g.getParameterModels().stream())
-            .collect(toList());
-
-        handleNestedParameters(ComponentModel.this, ComponentModel.this.directChildrenStream()
-            .filter(childComp -> childComp != ComponentModel.this),
-                               nestedComponents, extensionModelHelper,
-                               model,
-                               parameterModel -> parameterModelFilter.test(parameterModel)
-                                   && !inlineGroupedParameters.contains(parameterModel));
-      }
-
-    });
-
-    // Last resort to try to find a matching metadata type for this component
-    if (!getModel(HasStereotypeModel.class).isPresent()) {
-      extensionModelHelper.findMetadataType(getType())
```

Comment:
How did you do now with these cases? This was relying on the Type (class) to build the metadataTypeModelAdapter for the parameterAst... how does this work from now on?

Reply:
I moved this here: https://github.com/mulesoft/mule/pull/8864/files#diff-343feb8864aee8603a8a65aa05f549a3R162